### DISCUSSION TITLE
nnn + tmux + preview-tui = blur

### DISCUSSION THREAD

**Discussion by danielb2** [jarun/nnn#1713] [Q&A]
hi, just using nnn + preview-tui I have no problems previewing images, but as soon as I do this in a tmux window, resulting images are blurry. What's causing this? Can I fix it?

- Terminal: Kitty

so it should be using icat to view

<img width="757" alt="Screenshot 2023-08-18 at 2 32 42 PM" src="https://github.com/jarun/nnn/assets/26678/808f3c18-09ef-4977-ab70-f7b22f499eb8">

<figure>example using inside tmux</figure>
<img width="754" alt="Screenshot 2023-08-18 at 2 32 31 PM" src="https://github.com/jarun/nnn/assets/26678/e6c034e0-e4c1-407b-a4a7-f4fcaf1f491f">
<figure>example not using tmux</figure>

**Comment by luukvbaal** [comment]
`preview-tui` does not currently facilitate this use-case (kitty itself can multiplex so why use tmux?).
However if `icat` works inside of a `tmux` session I guess we could support this by setting `NNN_PREVIEWIMGPROG` to `"icat"`.
Please let us know if it works and open a PR if you can.

**Comment by danielb2** [comment]
it already has this line

```
    if [ "$NNN_TERMINAL" = "kitty" ] && [ -z "$NNN_PREVIEWIMGPROG" ]; then
        kitty +kitten icat --silent --scale-up --place "$1"x"$2"@0x0 --transfer-mode=stream --stdin=no "$3" &
```

**Comment by danielb2** [comment]
@luukvbaal any ideas?

**Reply by luukvbaal** [reply]
Yes like I said you could test whether forcing the use of the icat kitten while in a tmux session will work. Something like:
```bash
if [ -n "$KITTY_LISTEN_ON" ] && { [ -z "$NNN_PREVIEWIMGPROG" ] || [ "$NNN_PREVIEWIMGPROG" = "icat" ] ;}; then
    kitty +kitten icat --silent --scale-up --place "$1"x"$2"@0x0 --transfer-mode=stream --stdin=no "$3" &
...
```
Kitty users could set `NNN_PREVIEWIMGPROG` to `icat` which would then be used even inside a tmux session. I don't know whether it works though...

If it doesn't, I don't think there is a resolution. `preview-tui` correctly detects the tmux session and uses `catimg/viu/chafa` to display an (indeed pixelated) image.

**Reply by danielb2** [reply]
looks like there's a few different things going on. kitty doesn't play well with viu + tmux. ( https://github.com/atanunq/viu mentions half blocks ). this isn't just viu, but catimg etc all have the same problem with kitty

then icat itself seems to have problems, but only if the tmux session is _detched_ using `-d`. however, removing the -d focuses the wrong window and has another set of problems. 

Since Kovid is very responsive, I'm guessing the approach is to alert him of the icat issue and fix that which I will.

Can be reproduced by just doing:

`tmux split-window -d -h 'kitty +kitten icat foo.jpg; sleep 1'`

which doesn't work

while

`tmux split-window -h 'kitty +kitten icat foo.jpg; sleep 1'` does work

**Reply by danielb2** [reply]
https://github.com/kovidgoyal/kitty/issues/6573

**Reply by danielb2** [reply]
I suspected this, but not possible.

**Reply by danielb2** [reply]
I suppose one way is to use kitty window instead of tmux window? it seems to work fine in `screen` but I can't seem to get it to work in tmux

**Comment by danielb2** [comment]
btw @luukvbaal have you looked into [timg](https://github.com/hzeller/timg)? works really well with your previwer and movie files too
